src/corehost/corehost.cpp

```diff
@@ -333,6 +337,51 @@ void buffering_trace_writer(const pal::char_t* message)
 {
     g_buffered_errors.append(message).append(_X("\n"));
 }
+
+// Determines if the current module (should be the apphost.exe) is marked as Windows GUI application
+// in case it's not a GUI application (so should be CUI) or in case of any error the function returns false.
+bool get_windows_graphical_user_interface_bit()
+{
+    // The first two bytes of a PE file are a constant signature.
+    const UINT16 PEFileSignature = 0x5A4D;
+
+    // The offset of the PE header pointer in the DOS header.
+    const int PEHeaderPointerOffset = 0x3C;
+
+    // The offset of the Subsystem field in the PE header.
+    const int SubsystemOffset = 0x5C;
+
+    // The value of the sybsystem field which indicates Windows GUI (Graphical UI)
+    const UINT16 WindowsGUISubsystem = 0x2;
+
+    HMODULE module = ::GetModuleHandleW(NULL);
+    MODULEINFO module_info;
+    if (!::GetModuleInformation(::GetCurrentProcess(), module, &module_info, sizeof(module_info)))
```

Comment:
This is unnecessary. You  can just cast `HMODULE` to `void*` to get the address.

Reply:
I could not find anywhere in the docs that it's guaranteed that HMODULE is the base address. But I guess windows will never break that anyway.

Reply:
Anyway - I need the call to get the size to make sure we don't overrun any buffers. I know it should never happen (since otherwise OS would not load the module), but I'd rather keep this code itself safe.

Reply:
You have to trust Windows that it validated the PE file headers while loading the image. Otherwise, you cannot really make this safe. E.g. the PE file can be mapped with holes - checking the length does not guarantee that accessing the memory will succeed.

Here is equivalent code from CoreCLR where we trust Windows that it validated the PE file: https://github.com/dotnet/coreclr/blob/030a3ea9b8dbeae89c90d34441d4d9a1cf4a7de6/src/inc/pedecoder.inl#L116

`GetModuleInformation` is diagnostic kind of API. It does not feel right calling it during startup.

Reply:
OK - makes sense - removed the range checks and all the code around it.